src/main/java/com/unstoppabledomains/resolution/Resolution.java

```diff
@@ -266,5 +279,25 @@ public Resolution build() {
         private void setProvider(IProvider provider) {
             this.provider = provider;
         }
+
+        /**
+         * Makes a call via provider to the blockchainProviderUrl and returns the networkId
+         * @param blockchainProviderUrl
+         * @return Network object or null if couldn't retrive the network
+         */
+        private Network getNetworkId(String blockchainProviderUrl) {
+            JsonObject body = new JsonObject();
+            body.addProperty("jsonrpc", "2.0");
+            body.addProperty("method", "net_version");
+            body.add("params", new JsonArray());
+            body.addProperty("id", 67);
+            try {
+                JsonObject response = provider.request(blockchainProviderUrl, body);
+                return Network.getByChainId(response.get("result").getAsInt());
+            } catch(Exception e) {
+                e.printStackTrace();
+                return null;
```

Comment:
Should we allow to use our API if a configuration failed? I'd break the flow by throwing an exception since it is a vital thing for our lib to run properly. Correct me, if I'm wrong.